openedx/core/lib/block_cache/block_structure.py

```diff
@@ -0,0 +1,588 @@
+
+            value (any picklable type) - The value to associate with the
+                given key for the given transformer's data.
+        """
+        self._transformer_data[transformer.name()][key] = value
+
+    def get_transformer_block_data(self, usage_key, transformer, key=None, default=None):
+        """
+        Returns the value associated with the given key for the given
+        transformer for the block identified by the given usage_key;
+        returns default if not found.
+
+        Arguments:
+            usage_key (UsageKey) - Usage key of the block whose
+                transformer data is requested.
+
+            transformer (BlockStructureTransformer) - The transformer
+                whose dictionary data is requested.
+
+            key (string) - A dictionary key to the transformer's data
+                that is requested.
+
+            default (any type) - The value to return if a dictionary
+                entry is not found.
+        """
+        block_data = self._block_data_map.get(usage_key)
+        if not block_data:
+            return default
+        else:
+            transformer_data = block_data._transformer_data.get(transformer.name(), {})
+            if key:
+                return transformer_data.get(key, default)
+            else:
+                return transformer_data or default
+
+    def set_transformer_block_data(self, usage_key, transformer, key, value):
+        """
+        Updates the given transformer's data dictionary with the given
+        key and value for the block identified by the given usage_key.
+
+        Arguments:
+            usage_key (UsageKey) - Usage key of the block whose
+                transformer data is to be updated.
+
+            transformer (BlockStructureTransformer) - The transformer
+                whose data is to be updated.
+
+            key (string) - A dictionary key to the transformer's data.
+
+            value (any picklable type) - The value to associate with the
+                given key for the given transformer's data for the
+                requested block.
+        """
+        self._block_data_map[usage_key]._transformer_data[transformer.name()][key] = value
+
+    def remove_transformer_block_data(self, usage_key, transformer, key):
+        """
+        Deletes the given transformer's data dictionary value for the
+        given key for the block identified by the given usage_key.
+
+        Arguments:
+            usage_key (UsageKey) - Usage key of the block whose
+                transformer data is to be deleted.
+
+            transformer (BlockStructureTransformer) - The transformer
+                whose data entry is to be deleted.
+
+            key (string) - The key to the transformer's dictionary entry
+                that is to be deleted.
+        """
+        self._block_data_map[usage_key]._transformer_data.get(transformer.name(), {}).pop(key, None)
```

Comment:
When we are asked to remove the transformer data, we don't want to create any data if it doesn't exist.